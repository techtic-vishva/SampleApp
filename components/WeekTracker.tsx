import { Ionicons } from '@expo/vector-icons'
import React from 'react'
import { Dimensions, Pressable } from 'react-native'
import { View, Text } from './Themed'
import DateCircle from './DateCircle'

type CalendarEntry = {
	date: string
	isOvernight: boolean
	progress: number
}

const size = Dimensions.get('window').width * 0.1
const r = size * 0.4

function Day({
	label,
	entry,
	selected,
	onPress
}: {
	label: string
	entry: CalendarEntry
	selected: boolean
	onPress: (dateString: string) => void
}) {
	return (
		<Pressable
			onPress={() => onPress(entry.date)}
			style={{
				flex: 1,
				display: 'flex',
				justifyContent: 'center',
				alignItems: 'center',
				backgroundColor: 'transparent'
			}}>
			<View style={{ height: 12, backgroundColor: 'transparent' }}>
				{entry?.isOvernight && <Ionicons name={'ios-moon'} size={9} color={'white'} />}
			</View>
			<DateCircle progress={entry?.progress || 0} lineWidth={3} widthPercent={0.1} />
			<View
				style={{
					position: 'absolute',
					width: size,
					height: size,
					backgroundColor: 'transparent',
					alignItems: 'center',
					justifyContent: 'center',
					display: 'flex',
					paddingLeft: 1
				}}>
				<Text style={{ fontFamily: 'objektiv-semi-bold' }}>{label}</Text>
			</View>
			<View style={{ height: 12, backgroundColor: 'transparent' }}>
				{selected && <Ionicons name={'caret-up'} size={9} color={'white'} style={{ paddingTop: 2 }} />}
			</View>
		</Pressable>
	)
}

const EmptyDataEntry: CalendarEntry = {
	date: '0000-00-00',
	isOvernight: false,
	progress: 0
}

const EmptyData = [
	EmptyDataEntry,
	EmptyDataEntry,
	EmptyDataEntry,
	EmptyDataEntry,
	EmptyDataEntry,
	EmptyDataEntry,
	EmptyDataEntry
]

export default function WeekTracker({
	selectedDate,
	data,
	onPress,
	onTouchStart,
	onTouchEnd
}: {
	selectedDate: string
	data?: CalendarEntry[]
	onPress: (dateString: string) => void
	onTouchStart: (e: any) => void
	onTouchEnd: (e: any) => void
}) {
	if (!Array.isArray(data) || data.length !== 7) data = EmptyData

	return (
		<View
			onTouchStart={onTouchStart}
			onTouchEnd={onTouchEnd}
			style={{
				width: '90%',
				display: 'flex',
				flexDirection: 'row',
				height: '10%',
				justifyContent: 'space-between',
				marginTop: 10,
				backgroundColor: 'transparent'
			}}>
			<Day label="M" entry={data[0]} selected={data[0]?.date?.substring(0, 10) === selectedDate} onPress={onPress} />
			<Day label="T" entry={data[1]} selected={data[1]?.date?.substring(0, 10) === selectedDate} onPress={onPress} />
			<Day label="W" entry={data[2]} selected={data[2]?.date?.substring(0, 10) === selectedDate} onPress={onPress} />
			<Day label="T" entry={data[3]} selected={data[3]?.date?.substring(0, 10) === selectedDate} onPress={onPress} />
			<Day label="F" entry={data[4]} selected={data[4]?.date?.substring(0, 10) === selectedDate} onPress={onPress} />
			<Day label="S" entry={data[5]} selected={data[5]?.date?.substring(0, 10) === selectedDate} onPress={onPress} />
			<Day label="S" entry={data[6]} selected={data[6]?.date?.substring(0, 10) === selectedDate} onPress={onPress} />
		</View>
	)
}
