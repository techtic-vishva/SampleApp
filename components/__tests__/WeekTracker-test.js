import { format } from 'date-fns'
import * as React from 'react'
import renderer from 'react-test-renderer'

import WeekTracker from '../WeekTracker'

it('render when passing the props', () => {
	const tree = renderer
		.create(
			<WeekTracker
				selectedDate={format(new Date(), 'yyyy-MM-dd')}
				data={[]}
				onPress={(dateString) => {
					console.log(dateString)
				}}
				onTouchEnd={{}}
				onTouchStart={{}}
			/>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
