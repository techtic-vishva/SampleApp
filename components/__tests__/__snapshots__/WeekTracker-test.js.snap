// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when passing the props 1`] = `
<View
  onTouchEnd={Object {}}
  onTouchStart={Object {}}
  style={
    Array [
      Object {
        "backgroundColor": "#0b0805",
      },
      Object {
        "backgroundColor": "transparent",
        "display": "flex",
        "flexDirection": "row",
        "height": "10%",
        "justifyContent": "space-between",
        "marginTop": 10,
        "width": "90%",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
    <RNSVGSvgView
      bbHeight={75}
      bbWidth={75}
      focusable={false}
      height={75}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "270deg",
                },
              ],
            },
            Object {},
          ],
          Object {
            "flex": 0,
            "height": 75,
            "width": 75,
          },
        ]
      }
      width={75}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
            ]
          }
          r={30}
          stroke={4282196484}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={null}
          strokeWidth={3}
        />
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
              "strokeLinecap",
            ]
          }
          r={30}
          stroke={4293755693}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={188.49555921538757}
          strokeLinecap={1}
          strokeWidth={3}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "display": "flex",
            "height": 75,
            "justifyContent": "center",
            "paddingLeft": 1,
            "position": "absolute",
            "width": 75,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "fontFamily": "objektiv-semi-bold",
            },
          ]
        }
      >
        M
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
    <RNSVGSvgView
      bbHeight={75}
      bbWidth={75}
      focusable={false}
      height={75}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "270deg",
                },
              ],
            },
            Object {},
          ],
          Object {
            "flex": 0,
            "height": 75,
            "width": 75,
          },
        ]
      }
      width={75}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
            ]
          }
          r={30}
          stroke={4282196484}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={null}
          strokeWidth={3}
        />
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
              "strokeLinecap",
            ]
          }
          r={30}
          stroke={4293755693}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={188.49555921538757}
          strokeLinecap={1}
          strokeWidth={3}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "display": "flex",
            "height": 75,
            "justifyContent": "center",
            "paddingLeft": 1,
            "position": "absolute",
            "width": 75,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "fontFamily": "objektiv-semi-bold",
            },
          ]
        }
      >
        T
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
    <RNSVGSvgView
      bbHeight={75}
      bbWidth={75}
      focusable={false}
      height={75}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "270deg",
                },
              ],
            },
            Object {},
          ],
          Object {
            "flex": 0,
            "height": 75,
            "width": 75,
          },
        ]
      }
      width={75}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
            ]
          }
          r={30}
          stroke={4282196484}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={null}
          strokeWidth={3}
        />
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
              "strokeLinecap",
            ]
          }
          r={30}
          stroke={4293755693}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={188.49555921538757}
          strokeLinecap={1}
          strokeWidth={3}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "display": "flex",
            "height": 75,
            "justifyContent": "center",
            "paddingLeft": 1,
            "position": "absolute",
            "width": 75,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "fontFamily": "objektiv-semi-bold",
            },
          ]
        }
      >
        W
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
    <RNSVGSvgView
      bbHeight={75}
      bbWidth={75}
      focusable={false}
      height={75}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "270deg",
                },
              ],
            },
            Object {},
          ],
          Object {
            "flex": 0,
            "height": 75,
            "width": 75,
          },
        ]
      }
      width={75}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
            ]
          }
          r={30}
          stroke={4282196484}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={null}
          strokeWidth={3}
        />
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
              "strokeLinecap",
            ]
          }
          r={30}
          stroke={4293755693}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={188.49555921538757}
          strokeLinecap={1}
          strokeWidth={3}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "display": "flex",
            "height": 75,
            "justifyContent": "center",
            "paddingLeft": 1,
            "position": "absolute",
            "width": 75,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "fontFamily": "objektiv-semi-bold",
            },
          ]
        }
      >
        T
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
    <RNSVGSvgView
      bbHeight={75}
      bbWidth={75}
      focusable={false}
      height={75}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "270deg",
                },
              ],
            },
            Object {},
          ],
          Object {
            "flex": 0,
            "height": 75,
            "width": 75,
          },
        ]
      }
      width={75}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
            ]
          }
          r={30}
          stroke={4282196484}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={null}
          strokeWidth={3}
        />
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
              "strokeLinecap",
            ]
          }
          r={30}
          stroke={4293755693}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={188.49555921538757}
          strokeLinecap={1}
          strokeWidth={3}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "display": "flex",
            "height": 75,
            "justifyContent": "center",
            "paddingLeft": 1,
            "position": "absolute",
            "width": 75,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "fontFamily": "objektiv-semi-bold",
            },
          ]
        }
      >
        F
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
    <RNSVGSvgView
      bbHeight={75}
      bbWidth={75}
      focusable={false}
      height={75}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "270deg",
                },
              ],
            },
            Object {},
          ],
          Object {
            "flex": 0,
            "height": 75,
            "width": 75,
          },
        ]
      }
      width={75}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
            ]
          }
          r={30}
          stroke={4282196484}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={null}
          strokeWidth={3}
        />
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
              "strokeLinecap",
            ]
          }
          r={30}
          stroke={4293755693}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={188.49555921538757}
          strokeLinecap={1}
          strokeWidth={3}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "display": "flex",
            "height": 75,
            "justifyContent": "center",
            "paddingLeft": 1,
            "position": "absolute",
            "width": 75,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "fontFamily": "objektiv-semi-bold",
            },
          ]
        }
      >
        S
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
    <RNSVGSvgView
      bbHeight={75}
      bbWidth={75}
      focusable={false}
      height={75}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Array [
            Object {
              "transform": Array [
                Object {
                  "rotate": "270deg",
                },
              ],
            },
            Object {},
          ],
          Object {
            "flex": 0,
            "height": 75,
            "width": 75,
          },
        ]
      }
      width={75}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
            ]
          }
          r={30}
          stroke={4282196484}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={null}
          strokeWidth={3}
        />
        <RNSVGCircle
          cx={37.5}
          cy={37.5}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeDashoffset",
              "strokeLinecap",
            ]
          }
          r={30}
          stroke={4293755693}
          strokeDasharray={
            Array [
              188.49555921538757,
              188.49555921538757,
            ]
          }
          strokeDashoffset={188.49555921538757}
          strokeLinecap={1}
          strokeWidth={3}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "display": "flex",
            "height": 75,
            "justifyContent": "center",
            "paddingLeft": 1,
            "position": "absolute",
            "width": 75,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "fontFamily": "objektiv-semi-bold",
            },
          ]
        }
      >
        S
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#0b0805",
          },
          Object {
            "backgroundColor": "transparent",
            "height": 12,
          },
        ]
      }
    />
  </View>
</View>
`;
